### ISSUE TITLE
Preview page randomly wont display

### ISSUE THREAD

**Issue by maddisondesigns** [WordPress/gutenberg#4625]
## Issue Overview
Tried to preview a page that I'd previously created and all it showed was "Please wait…Generating preview."

![gutenberg-previewnotdisplaying](https://user-images.githubusercontent.com/2078245/35200549-c19b696e-ff64-11e7-9003-9b7c71d9729c.png)

## Steps to Reproduce (for bugs)
1. Open previously created page
2. Click Preview icon

## Expected Behavior
I expect to be able to preview my pages EVERY TIME

## Current Behavior
Gutenberg randomly wont let you preview the page.

_Firefox Quantum 57.0.4 (64-bit)
Gutenberg 2.0.0
WordPress 4.9.2_

**Comment by pento** [comment]
I haven't been able to reproduce this exactly as described, but I could reproduce the same behaviour like so:

- Open an existing post
- Click in the Post Title
- Add a character
- Press the preview button

@maddisondesigns: When you see this happening (without making any changes to the post), do you recall if the "Update" button is enabled or disabled? Also, do you have any plugins enabled on this site?

**Comment by maddisondesigns** [comment]
I have a feeling the _Update_ button was enabled because I was retesting multiple issues that I'd raised and I think on this occasion I had click on the Text Columns, changed the number of columns, broke the Text Columns block and then tried to preview.

If I do those steps now (Open page, Click Text Columns block, Change the number of columns, Click Prewiew), I can reliably replicate the issue.

If I also open the page, add a Paragraph Block (with or without text) and then try to preview, the same thing occurs.

Even by opening the page, adding a character to an existing Paragraph block and then clicking Preview, produces the same result.

Currently activated plugins are:
Advanced Custom Fields (Free)
WooCommerce
Yoast SEO

But I've also just deactivated every plugin (except Gutenberg) and can reproduce the same issue by performing the above steps

**Comment by maddisondesigns** [comment]
If I simply open a page and then click preview (without doing anything else), the preview seems to work ok

**Comment by pento** [comment]
Cool, thanks for the extra info. This should be fixed by #4218, which adds autosaving support for published posts, amongst a bunch of other autosaving improvements. (The preview button needs to make a successful autosave before it can open the preview URL.)

**Comment by maddisondesigns** [comment]
So I'm not going to be able to open a page and click preview straight away? That'll be really annoying!

**Comment by pento** [comment]
> This should be fixed by #4218

**Comment by kauaicreative** [comment]
Not fixed. Preview only works for me if the document has first been saved.

**Comment by maddisondesigns** [comment]
Seconded. This issue still exists and is easily reproducible by performing the steps I mentioned above (Open page, Click Text Columns block, Change the number of columns, Click Preview)

**Comment by pento** [comment]
This was closed as a duplicate of #4218, which hasn't been merged yet.

**Comment by maddisondesigns** [comment]
Can you be a little clearer next time then please. Saying it's Fixed is not helpful to us when it hasn't actually been merged. For people who are testing with the plugin, it's only 'Fixed' when it's actually merged in the plugin and we can test it.

**Comment by seezee** [comment]
I just installed the Gutenberg plugin. Same problem, except it's every time. I can't get the preview to work at all. In any case, converting my page completely borked the layout. I'll be installing the classic editor plugin when they roll this into core — it's a PITA.

**Comment by maddisondesigns** [comment]
Can someone please reopen this as it's still not fixed. Using Gutenberg v2.7.0 I can easily replicate the issue by performing the steps mentioned above.
(Open page, Click Text Columns block, Change the number of columns, Click Preview)

**Comment by pento** [comment]
I appreciate that it's frustrating that previewing published posts doesn't work, but that's not the bug here. The issue is that the autosave API hasn't been implemented yet, previewing published posts cannot work without that. The particular behaviour reported in this issue is a single aspect of a much larger problem.

We can certainly reopen this issue if, after #6257 is merged, previewing published posts still doesn't work.
